AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSGraphQLSubscriptionOperation.swift

```diff
@@ -174,18 +174,41 @@ final public class AWSGraphQLSubscriptionOperation<R: Decodable>: GraphQLSubscri
     }
 
     private func onSubscriptionFailure(_ error: Error) {
-        let errorDescription = "Subscription item event failed with error"
+        var errorDescription = "Subscription item event failed with error"
         if case let ConnectionProviderError.subscription(_, payload) = error,
            let errors = payload?["errors"] as? AppSyncJSONValue,
            let graphQLErrors = try? GraphQLErrorDecoder.decodeAppSyncErrors(errors) {
+
+            if AWSGraphQLSubscriptionOperation.anyGraphQLErrorIsUnauthorized(graphQLErrors) {
+                errorDescription += ": \(APIError.UnauthorizedMessageString)"
+            }
+
             let graphQLResponseError = GraphQLResponseError<R>.error(graphQLErrors)
             dispatch(result: .failure(APIError.operationError(errorDescription, "", graphQLResponseError)))
             finish()
             return
+        } else if case ConnectionProviderError.unauthorized = error {
+            errorDescription += ": \(APIError.UnauthorizedMessageString)"
         }
 
         dispatch(result: .failure(APIError.operationError(errorDescription, "", error)))
         finish()
     }
 
+    private static func anyGraphQLErrorIsUnauthorized(_ graphQLErrors: [GraphQLError]) -> Bool {
```

Comment:
not critical, but another option could be to make this an extension of `Array where Element == GraphQLError` so you could `errors.hasUnauthorizedError()`